lab_1/main.py

```diff
@@ -104,10 +154,32 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if not (isinstance(unknown_profile, dict)
+            and isinstance(profile_to_compare, dict)
+            and isinstance(top_n, int)):
+        return None
+    top_n_words_unknown_profile = get_top_n_words(unknown_profile['freq'], top_n)
+    top_n_words_profile_to_compare = get_top_n_words(profile_to_compare['freq'], top_n)
+    common = []
+    for i in top_n_words_profile_to_compare:
+        if i in top_n_words_unknown_profile:
+            common.append(i)
+    score = float(len(common) / len(top_n_words_unknown_profile))
```

Comment:
why float?

Reply:
My bad, in function compare_profiles we had to get a floating-point object (it was compulsory), but here, in function compare_profiles_advanced it is not necessarily (it may be integer).